multiaccounts/settings/database.go

```diff
@@ -0,0 +1,299 @@
+type Database struct {
+	db        *sql.DB
+	SyncQueue chan SyncSettingField
+}
+
+// MakeNewDB ensures that a singleton instance of Database is returned per sqlite db file
+func MakeNewDB(db *sql.DB) (*Database, error) {
+	filename, err := appdatabase.GetDBFilename(db)
+	if err != nil {
+		return nil, err
+	}
+
+	d := &Database{
+		db:        db,
+		SyncQueue: make(chan SyncSettingField, 100),
+	}
+
+	// An empty filename means that the sqlite database is held in memory
+	// In this case we don't want to restrict the instantiation
+	if filename == "" {
+		return d, nil
+	}
+
+	// Lock to protect the map from concurrent access
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	// init dbInstances if it hasn't been already
+	if dbInstances == nil {
+		dbInstances = map[string]*Database{}
+	}
+
+	// If we haven't seen this database file before make an instance
+	if _, ok := dbInstances[filename]; !ok {
+		dbInstances[filename] = d
+	}
+
+	return dbInstances[filename], nil
+}
+
+// TODO remove photoPath from settings
+func (db *Database) CreateSettings(s Settings, n params.NodeConfig) error {
+	tx, err := db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	_, err = tx.Exec(`
+INSERT INTO settings (
+  address,
+  currency,
+  current_network,
+  dapps_address,
+  eip1581_address,
+  installation_id,
+  key_uid,
+  keycard_instance_uid,
+  keycard_paired_on,
+  keycard_pairing,
+  latest_derived_path,
+  mnemonic,
+  name,
+  networks,
+  photo_path,
+  preview_privacy,
+  public_key,
+  signing_phrase,
+  wallet_root_address,
+  synthetic_id
+) VALUES (
+?,?,?,?,?,?,?,?,?,?,
+?,?,?,?,?,?,?,?,?,'id')`,
+		s.Address,
+		s.Currency,
+		s.CurrentNetwork,
+		s.DappsAddress,
+		s.EIP1581Address,
+		s.InstallationID,
+		s.KeyUID,
+		s.KeycardInstanceUID,
+		s.KeycardPAiredOn,
+		s.KeycardPairing,
+		s.LatestDerivedPath,
+		s.Mnemonic,
+		s.Name,
+		s.Networks,
+		s.PhotoPath,
+		s.PreviewPrivacy,
+		s.PublicKey,
+		s.SigningPhrase,
+		s.WalletRootAddress,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nodecfg.SaveConfigWithTx(tx, &n)
+}
+
+func (db *Database) getSettingFieldFromReactName(reactName string) (SettingField, error) {
+	for _, s := range SettingFieldRegister {
+		if s.GetReactName() == reactName {
+			return s, nil
+		}
+	}
+	return SettingField{}, errors.ErrInvalidConfig
+}
+
+func (db *Database) saveSetting(setting SettingField, value interface{}) error {
+	query := "UPDATE settings SET %s = ? WHERE synthetic_id = 'id'"
+	query = fmt.Sprintf(query, setting.GetDBName())
+	update, err := db.db.Prepare(query)
+	if err != nil {
+		return err
+	}
+
+	if setting.ValueHandler() != nil {
+		value, err = setting.ValueHandler()(value)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO(samyoul) this is ugly as hell need a more elegant solution
```

Comment:
I agree with this comment. There's a "settings_saveNodeConfig" RPC method. Maybe we could just remove the node config as a valid setting, and update mobile and desktop to use that rpc method instead of settings_saveSetting for updating the node config?  

Doing this change in desktop shouldnt take long, I'd say that it should be the same case in mobile.

Reply:
Ah, you've pin pointed my problem. Node config now operates outside of the normal settings flow, because it isn't classed as a setting anymore.

Yes, I agree with your suggestion, I'll make the change on mobile, and removed node config from the settings API.